crates/compiler/can/src/desugar.rs

```diff
@@ -143,6 +143,218 @@ pub fn desugar_defs<'a>(
     }
 }
 
+fn desugar_defs_node_suffixed<'a>(
+    arena: &'a Bump,
+    loc_expr: &'a Loc<Expr<'a>>,
+) -> &'a Loc<Expr<'a>> {
+    match loc_expr.value {
+        Defs(defs, loc_ret) => {
+            match defs.search_suffixed_defs() {
+                None => loc_expr,
+                Some((tag_index, value_index)) => {
+                    if defs.value_defs.len() == 1 {
+                        // We have only one value_def and it must be Suffixed
+                        // replace Defs with an Apply(Task.await) and Closure of loc_return
+
+                        assert!(
+                            value_index == 0,
+                            "we have only one value_def and so it must be Suffixed "
+                        );
+
+                        // Unwrap Suffixed def within Apply, and the pattern so we can use in the call to Task.await
+                        let (suffixed_sub_apply_loc, pattern) = unwrap_suffixed_def_and_pattern(
+                            arena,
+                            loc_expr.region,
+                            defs.value_defs[0],
+                        );
+
+                        // Create Closure for the result of the recursion,
+                        // use the pattern from our Suffixed Def as closure arument
+                        let closure_expr = Closure(arena.alloc([*pattern]), loc_ret);
+
+                        // Apply arguments to Task.await, first is the unwrapped Suffix expr second is the Closure
+                        let mut task_await_apply_args: Vec<&'a Loc<Expr<'a>>> = Vec::new_in(arena);
+
+                        task_await_apply_args
+                            .push(arena.alloc(Loc::at(loc_expr.region, suffixed_sub_apply_loc)));
+                        task_await_apply_args
+                            .push(arena.alloc(Loc::at(loc_expr.region, closure_expr)));
+
+                        arena.alloc(Loc::at(
+                            loc_expr.region,
+                            Apply(
+                                arena.alloc(Loc {
+                                    region: loc_expr.region,
+                                    value: Var {
+                                        module_name: ModuleName::TASK,
+                                        ident: "await",
+                                    },
+                                }),
+                                arena.alloc(task_await_apply_args),
+                                CalledVia::Space,
```

Comment:
Doesn't have to be in this PR, but I think we should add a `CalledVia::BangSuffix` and use it here, so we can tell later on in error messages that this `Task.await` was generated during desugaring.